src/config/spark.py

```diff
@@ -21,11 +21,11 @@ class SparkConfig(WithLogging):
     def __init__(
         self,
         s3_info: Optional[S3ConnectionInfo],
-        service_account_info: Optional[ServiceAccountInfo],
+        service_account_info: Optional[SparkServiceAccount],
     ):
         self.s3_info = s3_info
-        self.namespace = service_account_info.namespace
-        self.service_account = service_account_info.service_account
+        self.namespace = service_account_info.namespace if service_account_info else ""
+        self.service_account = service_account_info.service_account if service_account_info else ""
```

Comment:
nit: I'd rather prefer these be null in case the info is not available yet. And also, I think that the logic of "get value or else null" would be better handled in the domain class `SparkServiceAccount` itself rather than here.

Reply:
I see your point, and I agree. I actually did a bit of refactoring for this. See [800356a](https://github.com/canonical/kyuubi-k8s-operator/pull/21/commits/800356a469ad80a21b99b7593f906f7afc09496a)